static/panes/cfg-view.ts

```diff
@@ -98,125 +85,79 @@ export class Cfg {
 
         const pickerEl = this.domRoot.find('.function-picker')[0] as HTMLInputElement;
-        this.functionPicker = new TomSelect(
-            pickerEl,
-            {
-                sortField: 'name',
-                valueField: 'name',
-                labelField: 'name',
-                searchField: ['name'],
-                dropdownParent: 'body',
-                plugins: ['input_autogrow'],
-                onChange: (val: string) => {
+        this.functionPicker = new TomSelect(pickerEl, {
+            sortField: 'name',
+            valueField: 'name',
+            labelField: 'name',
+            searchField: ['name'],
+            dropdownParent: 'body',
+            plugins: ['input_autogrow'],
+            onChange: (e: any) => {
+                // TomSelect says it's an Event, but we receive strings
+                const val = e as string;
+                if (val in this.functions) {
                     const selectedFn = this.functions[val];
-                    if (selectedFn) {
-                        this.currentFunc = val;
-                        this.showCfgResults({
-                            nodes: selectedFn.nodes,
-                            edges: selectedFn.edges,
-                        });
+                    this.currentFunc = val;
+                    this.showCfgResults({
+                        nodes: selectedFn.nodes,
+                        edges: selectedFn.edges,
+                    });
+                    if (selectedFn.nodes && selectedFn.nodes.length > 0) {
                         this.cfgVisualiser.selectNodes([selectedFn.nodes[0].id]);
-                        this.resize();
-                        this.saveState();
                     }
-                },
-            } as any
-            // The current version of tom-select has a very restrictive type definition
-            // that forces to pass the whole options object. This is a workaround to make it type check
-        );
-
-        this.paneRenaming = new PaneRenaming(this, state);
+                    this.resize();
+                    this.updateState();
+                }
+            },
+        });
 
-        this.initCallbacks();
         this.updateButtons();
+    }
+
+    override getInitialHTML(): string {
+        return $('#cfg').html();
+    }
+
+    override registerOpeningAnalyticsEvent(): void {
         ga.proxy('send', {
             hitType: 'event',
             eventCategory: 'OpenViewPane',
             eventAction: 'Cfg',
         });
     }
 
-    onCompileResult(compilerId: number, compiler: any, result: any) {
-        if (this.compilerId === compilerId) {
+    override onCompileResult(compilerId: number, compiler: any, result: any) {
+        if (this.compilerInfo.compilerId === compilerId) {
             let functionNames: string[] = [];
-            if (this.supportsCfg && !$.isEmptyObject(result.cfg)) {
+            if (compiler.supportsCfg && !$.isEmptyObject(result.cfg)) {
```

Comment:
Storing the supportsCfg flag locally was doing weird stuff. On pane creation, the code in `onCompiler` ran first and set the flag to true, but then by the time it would reach this point, it would be `false` again! So I decided to just use the up-to-date status from the passed compiler object